### PR TITLE
Add documentation for dataset_metadata.yaml

### PR DESCRIPTION
**Pull request by jklukas**
Should be helpful for efforts like https://github.com/mozilla/bigquery-etl/pull/2105 in the future

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -296,7 +296,10 @@ labels:
   incremental_export: false  # non-incremental JSON export writes all data to a single location
 ```
 
-### Publishing Datasets
+### Publishing a Table Publicly
+
+For background, see [Accessing Public Data](https://docs.telemetry.mozilla.org/cookbooks/public_data.html)
+on `docs.telemetry.mozilla.org`.
```
Comment (jklukas) [inline README.md:302]: Unrelated change for something I was confused by while reading through this file.

**Review by scholtzan — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -316,6 +319,45 @@ labels:
 - For each dataset, a `metadata.json` gets published listing all available files, for example: https://public-data.telemetry.mozilla.org/api/v1/tables/telemetry_derived/ssl_ratios/v1/files/metadata.json
 - The timestamp when the dataset was last updated is recorded in `last_updated`, e.g.: https://public-data.telemetry.mozilla.org/api/v1/tables/telemetry_derived/ssl_ratios/v1/last_updated
 
+Dataset Metadata
+---
+
+To provision a new BigQuery dataset for holding tables, you'll need to
+create a `dataset_metadata.yaml` which will cause the dataset to be
+automatically deployed.
```
Comment (whd) [inline README.md:327]: It's possibly worth mentioning that normal dataset deploy propagation to production takes 0.5-2 hours from when `generated-sql` is updated. There's currently no deploy notification to #dataops-alerts for this kind of deploy.
